odl/solvers/util/callback.py

```diff
@@ -325,12 +333,38 @@ def __init__(self, *args, **kwargs):
         self.kwargs = kwargs
         self.fig = kwargs.pop('fig', None)
         self.display_step = kwargs.pop('display_step', 1)
+        self.saveto = kwargs.pop('saveto', None)
+        self.suffix = kwargs.pop('suffix', 'png')
         self.iter = 0
 
+        if self.saveto is not None:
+            head, tail = os.path.split(self.saveto)
+            if not os.path.exists(head) or head == '':
+                raise ValueError('head of the `saveto` {} path not found'
+                                 ''.format(head))
+            if ((not tail.count('{') == tail.count('}'))):
+                raise ValueError('tail of `saveto` {} does not contain the '
+                                 ' same number of ``{`` as ``}``'.format(tail))
+            elif tail.count('{') == 1:
```

Comment:
IMO don't check at all. Just use the format as-is, the user's gonna get what she deserves. :-)